From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: UN1CA <unica@local>
Subject: [PATCH] Connect Cipher island to native SystemUI events with stock fallback

Upstream: sampple-korea https://github.com/sampple-korea/dynamicisland
Native integration generated by unica/mods/island/build.sh.

diff --git a/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController.smali b/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController.smali
@@ -212,6 +212,8 @@

     invoke-direct {p0}, Landroid/app/Service;-><init>()V

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V
+
     iput-object p1, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->ctx:Landroid/content/Context;

     iput-object p2, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->commonNotifCollection:Ldagger/Lazy;
@@ -760,6 +762,8 @@
 .method public final OaDataListChanged()V
     .locals 0

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->sendChipInfoForAppExitAnimation()V

     return-void
@@ -767,6 +771,8 @@

 .method public final add$1()V
     .locals 2
+
+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V

     const-string/jumbo v0, "{OngoingActivityController}"

@@ -1692,6 +1698,14 @@
 .method public final isChipExpandCondition()Z
     .locals 10

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->shouldReplace()Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    const/4 v0, 0x0
+    return v0
+
+    :cond_cipher_original
+
     sget-object v0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->INSTANCE:Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;

     invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
@@ -1803,6 +1817,8 @@
 .method public final killGhost()V
     .locals 6

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     sget-object v0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->INSTANCE:Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;

     iget-object p0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->commonNotifCollection:Ldagger/Lazy;
@@ -2039,6 +2055,8 @@

 .method public final onAsyncInflationFinished(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;Z)V
     .locals 11
+
+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V

     if-eqz p2, :cond_0

@@ -2933,6 +2951,8 @@
 .method public final remove(I)V
     .locals 4

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     sget-object v0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->INSTANCE:Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;

     invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
@@ -2998,6 +3018,8 @@
 .method public final removeOngoingActivityData(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController$DelayedEntryState;)V
     .locals 6

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->mOngoingCardController:Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingCardController;

     const/4 v1, 0x1
@@ -3226,6 +3248,13 @@

 .method public final sendChipInfoForAppExitAnimation()V
     .locals 14
+
+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->exitTargets(Ljava/lang/Object;)Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original

     const-string/jumbo v0, "sendChipInfoForAppExitAnimation check OA data"

@@ -3725,6 +3754,8 @@
 .method public final update(Ljava/lang/String;)V
     .locals 4

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     const-string/jumbo v0, "update("

     const-string v1, ")"
@@ -3786,6 +3817,8 @@
 .method public final update$1$1()V
     .locals 2

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     const-string/jumbo v0, "{OngoingActivityController}"

     const-string/jumbo v1, "update()"
@@ -3887,6 +3920,8 @@
 .method public final updateMediaChipData()V
     .locals 11

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->currentMediaOngoingActivityInfo:Lcom/android/systemui/statusbar/phone/ongoingactivity/MediaOngoingActivityInfo;

     if-nez v0, :cond_0
@@ -4265,6 +4300,8 @@
 .method public final updateMediaChipData(Landroid/graphics/drawable/Icon;Ljava/lang/String;ILandroid/app/PendingIntent;Ljava/lang/Boolean;)V
     .locals 9

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     iget-boolean v0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->isUpdateNotAllowed:Z

     const-string/jumbo v1, "{OngoingActivityController}"
@@ -4774,6 +4811,13 @@
 .method public final updateParentViewVisibility(Z)V
     .locals 6

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->hideParent(Ljava/lang/Object;)Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original
+
     iget-boolean v0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityController;->shouldHiddenByOtherChips:Z

     const/4 v1, 0x1
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController.smali b/smali_classes3/com/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController.smali
@@ -119,6 +119,8 @@

     invoke-direct {p0}, Ljava/lang/Object;-><init>()V

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V
+
     iput-object p1, p0, Lcom/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController;->scope:Lkotlinx/coroutines/CoroutineScope;

     iput-object p2, p0, Lcom/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController;->context:Landroid/content/Context;
@@ -536,6 +538,8 @@
 .method public final removeChipInfo()V
     .locals 4

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     iget-boolean v0, p0, Lcom/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController;->callChipAppeared:Z

     const/4 v1, 0x0
@@ -593,6 +597,8 @@

 .method public final sendStateChangeEvent()V
     .locals 7
+
+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V

     invoke-virtual {p0}, Lcom/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController;->hasOngoingCall()Z

@@ -1051,6 +1057,8 @@
 .method public final updateChip()V
     .locals 13

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController;->callNotificationInfo:Lcom/android/systemui/statusbar/phone/ongoingcall/SamsungOngoingCallController$CallNotificationInfo;

     if-nez v0, :cond_0
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastController.smali b/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastController.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastController.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastController.smali
@@ -101,6 +101,8 @@

     invoke-direct {p0}, Ljava/lang/Object;-><init>()V

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V
+
     iput-object p1, p0, Lcom/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastController;->context:Landroid/content/Context;

     iput-object p3, p0, Lcom/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastController;->scope:Lkotlinx/coroutines/CoroutineScope;
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastViewModel$SamsungCastChipBundle.smali b/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastViewModel$SamsungCastChipBundle.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastViewModel$SamsungCastChipBundle.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastViewModel$SamsungCastChipBundle.smali
@@ -37,6 +37,8 @@

     invoke-direct {p0}, Ljava/lang/Object;-><init>()V

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V
+
     iput-object p1, p0, Lcom/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastViewModel$SamsungCastChipBundle;->screenRecord:Lcom/android/systemui/statusbar/chips/ui/model/OngoingActivityChipModel;

     iput-object p2, p0, Lcom/android/systemui/statusbar/phone/ongoingcast/SamsungOngoingCastViewModel$SamsungCastChipBundle;->castToOtherDevice:Lcom/android/systemui/statusbar/chips/ui/model/OngoingActivityChipModel;
diff --git a/smali_classes3/com/android/systemui/shade/ShadeExpansionStateManager.smali b/smali_classes3/com/android/systemui/shade/ShadeExpansionStateManager.smali
--- a/smali_classes3/com/android/systemui/shade/ShadeExpansionStateManager.smali
+++ b/smali_classes3/com/android/systemui/shade/ShadeExpansionStateManager.smali
@@ -42,6 +42,8 @@

     invoke-direct {p0}, Ljava/lang/Object;-><init>()V

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V
+
     new-instance v0, Ljava/util/concurrent/CopyOnWriteArrayList;

     invoke-direct {v0}, Ljava/util/concurrent/CopyOnWriteArrayList;-><init>()V
diff --git a/smali_classes3/com/android/systemui/statusbar/StatusBarStateControllerImpl.smali b/smali_classes3/com/android/systemui/statusbar/StatusBarStateControllerImpl.smali
--- a/smali_classes3/com/android/systemui/statusbar/StatusBarStateControllerImpl.smali
+++ b/smali_classes3/com/android/systemui/statusbar/StatusBarStateControllerImpl.smali
@@ -108,6 +108,8 @@

     invoke-direct {p0}, Ljava/lang/Object;-><init>()V

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V
+
     new-instance p3, Ljava/util/ArrayList;

     invoke-direct {p3}, Ljava/util/ArrayList;-><init>()V
diff --git a/smali_classes4/com/android/systemui/statusbar/window/StatusBarWindowStateController.smali b/smali_classes4/com/android/systemui/statusbar/window/StatusBarWindowStateController.smali
--- a/smali_classes4/com/android/systemui/statusbar/window/StatusBarWindowStateController.smali
+++ b/smali_classes4/com/android/systemui/statusbar/window/StatusBarWindowStateController.smali
@@ -23,6 +23,8 @@
     .locals 1

     invoke-direct {p0}, Ljava/lang/Object;-><init>()V
+
+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->capture(Ljava/lang/Object;)V

     iput p1, p0, Lcom/android/systemui/statusbar/window/StatusBarWindowStateController;->thisDisplayId:I

diff --git a/smali_classes3/com/android/systemui/statusbar/phone/PhoneStatusBarViewController.smali b/smali_classes3/com/android/systemui/statusbar/phone/PhoneStatusBarViewController.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/PhoneStatusBarViewController.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/PhoneStatusBarViewController.smali
@@ -791,6 +791,8 @@
 .method public final onViewAttached()V
     .locals 11

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->attached(Ljava/lang/Object;)V
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/PhoneStatusBarViewController;->samsungExtLazy:Ldagger/Lazy;

     invoke-interface {v0}, Ldagger/Lazy;->get()Ljava/lang/Object;
@@ -1360,6 +1362,8 @@
 .method public onViewDetached()V
     .locals 6

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->detached(Ljava/lang/Object;)V
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/PhoneStatusBarViewController;->samsungExtLazy:Ldagger/Lazy;

     invoke-interface {v0}, Ldagger/Lazy;->get()Ljava/lang/Object;
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper.smali b/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper.smali
@@ -1494,6 +1494,8 @@
 .method public static notifyRankingStateChanged(Ljava/lang/String;)V
     .locals 6

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     invoke-static {p0}, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->getOngoingActivityDataByKey(Ljava/lang/String;)Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityData;

     move-result-object v0
@@ -1549,6 +1551,8 @@
 .method public static notifyRemoveItemNowbarObservers(Ljava/lang/String;)V
     .locals 8

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     sget-object v0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->nowbarObservers:Ljava/util/List;

     check-cast v0, Ljava/util/ArrayList;
@@ -1630,6 +1634,8 @@
 .method public static notifyUpdateItemNowbarObservers(Lcom/android/systemui/plugins/keyguardstatusview/NowBarItem;)V
     .locals 4

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     sget-object v0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->nowbarObservers:Ljava/util/List;

     check-cast v0, Ljava/util/ArrayList;
@@ -1896,6 +1902,8 @@
 .method public static removeOngoingActivityByKey(Ljava/lang/String;)V
     .locals 7

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     const-string v0, "findOngoingActivityDataIndex() noti id = "

     sget-object v1, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->TAG:Ljava/lang/String;
@@ -2547,6 +2555,8 @@
 .method public static updateOngoingList(Lcom/android/systemui/statusbar/NotificationLockscreenUserManager;Z)V
     .locals 7

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     new-instance v0, Ljava/util/LinkedList;

     invoke-direct {v0}, Ljava/util/LinkedList;-><init>()V
@@ -2892,6 +2902,8 @@
 .method public final setBaseActivityComponentName(Landroid/content/ComponentName;Lcom/android/systemui/statusbar/NotificationLockscreenUserManager;)V
     .locals 1

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->refresh()V
+
     sget-object p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/OngoingActivityDataHelper;->baseActivityComponentName:Landroid/content/ComponentName;

     invoke-virtual {p1, p0}, Landroid/content/ComponentName;->equals(Ljava/lang/Object;)Z
diff --git a/smali_classes3/com/android/systemui/statusbar/pipeline/shared/ui/binder/HomeStatusBarViewBinderImpl.smali b/smali_classes3/com/android/systemui/statusbar/pipeline/shared/ui/binder/HomeStatusBarViewBinderImpl.smali
--- a/smali_classes3/com/android/systemui/statusbar/pipeline/shared/ui/binder/HomeStatusBarViewBinderImpl.smali
+++ b/smali_classes3/com/android/systemui/statusbar/pipeline/shared/ui/binder/HomeStatusBarViewBinderImpl.smali
@@ -48,6 +48,13 @@
 .method public static adjustVisibility(Landroid/view/View;Lcom/android/systemui/statusbar/pipeline/shared/ui/model/VisibilityModel;)V
     .locals 5

+    invoke-static/range {p0 .. p1}, Ldev/android/dynamicisland/island/NativeBridge;->hideView(Landroid/view/View;Ljava/lang/Object;)Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original
+
     iget v0, p1, Lcom/android/systemui/statusbar/pipeline/shared/ui/model/VisibilityModel;->visibility:I

     iget-boolean p1, p1, Lcom/android/systemui/statusbar/pipeline/shared/ui/model/VisibilityModel;->shouldAnimateChange:Z
diff --git a/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController.smali b/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController.smali
--- a/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController.smali
+++ b/smali_classes3/com/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController.smali
@@ -430,6 +430,13 @@
 .method public final animateChipShow(Landroid/view/View;Z)V
     .locals 2

+    invoke-static/range {p1 .. p1}, Ldev/android/dynamicisland/island/NativeBridge;->hideCapsule(Landroid/view/View;)Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original
+
     new-instance v0, Lcom/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController$animateChipShow$1;

     const/4 v1, 0x0
@@ -448,6 +455,13 @@
 .method public final clockViewNudgeAnimation()V
     .locals 5

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->shouldReplace()Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController;->isLeftClockPosition:Lkotlin/jvm/functions/Function0;

     invoke-interface {v0}, Lkotlin/jvm/functions/Function0;->invoke()Ljava/lang/Object;
@@ -558,6 +572,13 @@
 .method public final handleOnGoingActivityChipAnimation(Z)V
     .locals 9

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->shouldReplace()Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original
+
     iget-object v0, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController;->onGoingActivityChip:Landroid/view/View;

     if-eqz v0, :cond_0
@@ -642,6 +663,13 @@
 .method public final handleOnGoingCallChipAnimation(Z)V
     .locals 7

+    invoke-static {}, Ldev/android/dynamicisland/island/NativeBridge;->shouldReplace()Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    return-void
+
+    :cond_cipher_original
+
     iget-object v2, p0, Lcom/android/systemui/statusbar/phone/ongoingactivity/ChipAnimationController;->onGoingCallChip:Landroid/view/View;

     if-eqz v2, :cond_1
diff --git a/smali_classes3/com/android/systemui/statusbar/notification/collection/ShadeListBuilder.smali b/smali_classes3/com/android/systemui/statusbar/notification/collection/ShadeListBuilder.smali
--- a/smali_classes3/com/android/systemui/statusbar/notification/collection/ShadeListBuilder.smali
+++ b/smali_classes3/com/android/systemui/statusbar/notification/collection/ShadeListBuilder.smali
@@ -569,6 +569,14 @@

 .method public static applyFilters(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;JLjava/util/List;)Z
     .locals 5
+
+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->hideNotification(Ljava/lang/Object;)Z
+    move-result v0
+    if-eqz v0, :cond_cipher_original
+    const/4 v0, 0x1
+    return v0
+
+    :cond_cipher_original

     invoke-interface {p3}, Ljava/util/List;->size()I

@@ -4233,6 +4241,9 @@
 .method public final filterNotifs(Ljava/util/Collection;Ljava/util/List;Ljava/util/List;)V
     .locals 7

+    invoke-static/range {p0 .. p0}, Ldev/android/dynamicisland/island/NativeBridge;->captureListBuilder(Ljava/lang/Object;)V
+    invoke-static/range {p1 .. p3}, Ldev/android/dynamicisland/island/NativeBridge;->snapshot(Ljava/util/Collection;Ljava/lang/Object;Ljava/lang/Object;)V
+
     const-string v0, "ShadeListBuilder.filterNotifs"

     invoke-static {v0}, Landroid/os/Trace;->beginSection(Ljava/lang/String;)V
diff --git a/smali_classes3/com/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl.smali b/smali_classes3/com/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl.smali
--- a/smali_classes3/com/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl.smali
+++ b/smali_classes3/com/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl.smali
@@ -1885,6 +1885,10 @@
 .method public final setEntryPinned(Lcom/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl$HeadsUpEntry;Lcom/android/systemui/statusbar/notification/headsup/PinnedStatus;Ljava/lang/String;)V
     .locals 9

+    invoke-static/range {p0 .. p2}, Ldev/android/dynamicisland/island/NativeBridge;->pinStatus(Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+    move-result-object p2
+    check-cast p2, Lcom/android/systemui/statusbar/notification/headsup/PinnedStatus;
+
     iget-object v0, p1, Lcom/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl$HeadsUpEntry;->mEntry:Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;

     iget-object v1, p0, Lcom/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl;->mLogger:Lcom/android/systemui/statusbar/notification/headsup/HeadsUpManagerLogger;
@@ -2324,7 +2328,7 @@
     return-void
 .end method

-.method public shouldHeadsUpBecomePinned(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;)Z
+.method public nativeOriginal_shouldHeadsUpBecomePinned(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;)Z
     .locals 4

     const/4 v0, 0x0
@@ -2415,6 +2419,19 @@
     return v0
 .end method

+.method public shouldHeadsUpBecomePinned(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;)Z
+    .locals 1
+
+    invoke-virtual {p0, p1}, Lcom/android/systemui/statusbar/notification/headsup/HeadsUpManagerImpl;->nativeOriginal_shouldHeadsUpBecomePinned(Lcom/android/systemui/statusbar/notification/collection/NotificationEntry;)Z
+    move-result v0
+    if-eqz v0, :return_stock
+    invoke-static {p0, p1}, Ldev/android/dynamicisland/island/NativeBridge;->headsUp(Ljava/lang/Object;Ljava/lang/Object;)Z
+    move-result v0
+    xor-int/lit8 v0, v0, 0x1
+    :return_stock
+    return v0
+.end method
+
 .method public final snooze()V
     .locals 9
